src/lib/build.ts

```diff
@@ -1,54 +1,79 @@
 import fs from 'fs'
 import process from 'process'
 
-import build from '@netlify/build'
+import build, { type OnEnd, type OnPostBuild } from '@netlify/build'
 // @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'toml... Remove this comment to see the full error message
 import tomlify from 'tomlify-j0.4'
+import type { OptionValues } from 'commander'
 
 import { getFeatureFlagsFromSiteInfo } from '../utils/feature-flags.js'
+import type { EnvironmentVariables } from '../utils/types.js'
 
 import { getBootstrapURL } from './edge-functions/bootstrap.js'
 import { featureFlags as edgeFunctionsFeatureFlags } from './edge-functions/consts.js'
 
-/**
- * The buildConfig + a missing cachedConfig
- * @typedef BuildConfig
- * @type {Parameters<import('@netlify/build/src/core/main.js')>[0] & {cachedConfig: any}}
- */
+export interface CachedConfig {
+  env: EnvironmentVariables
+  siteInfo: {
+    id?: string
+    account_id?: string
+    feature_flags?: Record<string, boolean | boolean | number>
+  }
+  // TODO(serhalp) Add remaining properties
+  [k: string]: unknown
+}
+
+interface DefaultConfig {
+  // TODO(serhalp) Add remaining properties
+  [k: string]: unknown
+}
+
+// TODO(serhalp) This is patching weak or missing properties from @netlify/build. Fix there instead.
+export type BuildConfig = Parameters<typeof build>[0] & {
+  cachedConfig: CachedConfig
+  defaultConfig: DefaultConfig
+  // It's possible this one is correct in @netlify/build. Remove and stop passing it if so.
+  accountId?: string
+  edgeFunctionsBootstrapURL: string
+}
+
+interface HandlerResult {
+  newEnvChanges?: Record<string, string>
+  configMutations?: Record<string, string>
+  status?: string
+}
+// The @netlify/build type incorrectly states a `void | Promise<void>` return type.
```

Comment:
😓 I guess this ended up growing and growing as I hacked away. I should probably upstream most of this instead of merging this as is, I guess.

Reply:
Seems fine to merge as is and upstream in another pass

Reply:
Yeah, it's also that in the coming days we'll be making several more changes like this. It would be a lot of back-and-forth to validate here + open PR in build + release PR in build + bump here + repeat. I'd rather batch it later.